packages/events/src/service/indexing/query.ts

```diff
@@ -77,124 +73,75 @@ function generateQuery(
     []
   )
 
-  const nameFieldIds = allEventFields
-    .filter((field) => field.type === FieldType.NAME)
-    .map((f) => f.id)
-
-  const addressFieldIds = allEventFields
-    .filter((field) => field.type === FieldType.ADDRESS)
-    .map((f) => f.id)
-
-  const alternateFieldMap = getAlternateFieldMap(eventConfigs)
-
-  const must = Object.entries(event)
-    // Exclude fields from Search query that are meant to
-    .filter(([fieldId]) => {
-      const fieldSearchConfig = eventConfigs
-        .flatMap((x) => x.advancedSearch)
-        .flatMap((s) => s.fields)
-        .find((f) => f.fieldId === fieldId)
-      if (
-        fieldSearchConfig?.fieldType === 'field' &&
-        fieldSearchConfig.excludeInSearchQuery
-      ) {
-        return false
-      }
-      return true
-    })
-    .map(([fieldId, search]) => {
-      const field = `declaration.${encodeFieldId(fieldId)}`
-
-      if (search.type === 'exact') {
-        const allIds = [fieldId, ...(alternateFieldMap[fieldId] ?? [])]
-
-        if (
-          fieldId in alternateFieldMap &&
-          alternateFieldMap[fieldId].length > 0
-        ) {
-          const queries: estypes.QueryDslQueryContainer[] = allIds.map((id) => {
-            if (addressFieldIds.includes(id)) {
-              const parsed = AddressFieldValue.safeParse(
-                JSON.parse(search.term)
-              )
-              if (parsed.success) {
-                return generateQueryForAddressField(
-                  `declaration.${encodeFieldId(id)}`,
-                  parsed.data
-                )
-              }
-            }
-
-            // For non-address or failed parse, return simple match
-            return {
-              match: {
-                [`declaration.${encodeFieldId(id)}`]: search.term
-              }
-            }
-          })
+  const must = Object.entries(event).map(([fieldId, search]) => {
+    const esFieldName = `declaration.${encodeFieldId(fieldId)}`
+    const field = getOrThrow(
+      allEventFields.find((f) => f.id === fieldId),
+      `Tried to search with a field id ${fieldId} but it is not found in event configuration`
+    )
 
-          return {
-            bool: { should: queries, minimum_should_match: 1 }
-          }
-        }
+    if (field.type === FieldType.ADDRESS) {
+      return generateQueryForAddressField(fieldId, search)
+    }
 
-        // default exact match
+    if (field.type === FieldType.NAME) {
+      if (search.type === 'fuzzy') {
         return {
           match: {
-            [field]: search.term
+            [`${esFieldName}.__fullname`]: {
+              query: search.term,
+              fuzziness: 'AUTO'
+            }
           }
         }
       }
+      return {
+        match: {
+          [esFieldName]: search.term
```

Comment:
even for exact search, shouldn't we be matching against the `__fullname` field? otherwise won't we we searching against an object?